src/referral/related_models/referral_invite.py

```diff
@@ -0,0 +1,88 @@
+from django.db import models
+from django.db.models import Sum
+
+from invite.models import Invitation
+from reputation.related_models.bounty import Bounty
+from researchhub.settings import BASE_FRONTEND_URL, REFERRAL_PROGRAM
+from researchhub_document.related_models.researchhub_unified_document_model import (
+    ResearchhubUnifiedDocument,
+)
+from utils.message import send_email_message
+
+JOIN_RH = "JOIN_RH"
+BOUNTY = "BOUNTY"
+
+INVITE_TYPE_CHOICES = [(JOIN_RH, JOIN_RH), (BOUNTY, BOUNTY)]
+
+
+class ReferralInvite(Invitation):
+    invite_type = models.CharField(
+        choices=INVITE_TYPE_CHOICES,
+        max_length=32,
+        default=JOIN_RH,
+        blank=False,
+        null=False,
+    )
+
+    unified_document = models.ForeignKey(
+        ResearchhubUnifiedDocument,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="referral_invites",
+    )
+
+    referral_first_name = models.CharField(max_length=256, blank=True, null=True)
+
+    referral_last_name = models.CharField(max_length=256, blank=True, null=True)
+
+    def _send_bounty_invitation(self):
+        inviter = self.inviter
```

Comment:
I'm tweaking this part a bit still based on template so some vars will be removed